src/main/java/com/azure/cosmos/examples/documentcrud/async/DocumentCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,563 @@
+        // happen while waiting for earlier processes to finish.
+        Thread.sleep(1000);
+
+        logger.info("replace and upsert done now...");
+        replaceDocumentWithConditionalEtagCheck();
+        readDocumentOnlyIfChanged();
+        // deleteDocument() is called at shutdown()
+
+    }
+
+    // Database Create
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Create database {} if not exists...", databaseName);
+
+        //  Create database if not exists
+        CosmosDatabaseResponse databaseResponse = client.createDatabaseIfNotExists(databaseName).block();
+        database = client.getDatabase(databaseResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container create
+    private void createContainerIfNotExists() throws Exception {
+        logger.info("Create container {} if not exists.", containerName);
+
+        // Create container if not exists - this is async but we block to make sure
+        // database and containers are created before sample runs the CRUD operations on
+        // them
+        CosmosContainerProperties containerProperties =
+                new CosmosContainerProperties(containerName, "/lastName");
+
+        // Provision throughput
+        ThroughputProperties throughputProperties = ThroughputProperties.createManualThroughput(400);
+
+        //  Create container with 200 RU/s
+        CosmosContainerResponse containerResponse = database.createContainerIfNotExists(containerProperties, throughputProperties).block();
+        container = database.getContainer(containerResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    private void createDocument() throws Exception {
+        logger.info("Create document {}", documentId);
+
+        // Define a document as a POJO (internally this
+        // is converted to JSON via custom serialization)
+
+        Family family = new Family();
+        family.setLastName(documentLastName);
+        family.setId(documentId);
+
+        // Insert this item as a document
+        // Explicitly specifying the /pk value improves performance.
+        // add subscribe() to make this async
+        container.createItem(family, new PartitionKey(family.getLastName()), new CosmosItemRequestOptions())
+                .doOnSuccess((response) -> {
+                    logger.info("inserted doc with id: {}", response.getItem().getId());
+                })
+                .doOnError((exception) -> {
+                    logger.error(
+                            "Exception. e: {}",
+                            exception.getLocalizedMessage(),
+                            exception);
+                }).subscribe();
+        logger.info("creating doc asynchronously...");
+    }
+
+    private void createDocuments(Flux<Family> families) throws Exception {
+        logger.info("Create documents {}", documentId);
```

Comment:
each family should have a different documentId I think? so the logging here is not very accurate